src/WorkflowCore/Services/WorkflowExecutor.cs

```diff
@@ -175,16 +176,23 @@ private void ProcessInputs(WorkflowInstance workflow, WorkflowStep step, IStepBo
             }
         }
 
-        private void ProcessOutputs(WorkflowInstance workflow, WorkflowStep step, IStepBody body)
+        private void ProcessOutputs(WorkflowInstance workflow, WorkflowStep step, IStepBody body, IStepExecutionContext context)
         {
             foreach (var output in step.Outputs)
             {
-                var member = (output.Target.Body as MemberExpression);
                 var resolvedValue = output.Source.Compile().DynamicInvoke(body);
                 var data = workflow.Data;
-                var property = data.GetType().GetProperty(member.Member.Name);
-                var convertedValue = Convert.ChangeType(resolvedValue, property.PropertyType);
-                property.SetValue(data, convertedValue);
+                var setter = ExpressionHelpers.CreateSetter(output.Target);
+                var convertedValue = Convert.ChangeType(resolvedValue, setter.Parameters.Last().Type);
```

Comment:
Don't we need similar changes for `ProcessInputs`?
Could you also add some test cases for the new functionality in the `WorkflowExecutorFixture`?

Reply:
No `ProcessInputs` does not need to be changed since it is already treated as a normal lambda. Only the Outputs are "special" in that they are just property names.

I was actually also thinking about changing both Inputs and Outputs to normal lambdas. That would enable getting rid of a lot of reflection and probably big performance improvements (at least for workflows with a lot of property and data movements).
[I started on it here.](https://github.com/dr-BEat/workflow-core/commit/7927c68f45387c1051b7315558f9808ac320bea6) I would love to hear you opinion about it.

Reply:
Oh, and I will also add some more tests to `WorkflowExecutorFixture`.

Reply:
I added the test.